adagrams/game.py

```diff
@@ -1,13 +1,166 @@
+        for i in range(count):
+            letter_list.append(letter)
+    print(letter_list)
+
+
+    pulled_letters = [] # pull random letters
+    for i in range(10): # max 10 letters
+        random_index = randint(0, len(letter_list) - 1) # randint generates random int between 2 values.
+        random_letter = letter_list.pop(random_index) # used pop method with help of Mikelle in office hours :)
+        pulled_letters.append(random_letter) # adds random letter to the list pulled letter with 
+
+    return pulled_letters
+
+# wave_02:
+# we need to check if word uses only letter from letter bank.
+# we need to keep in ind frequancies of each letter in both word and letter bank.
+# return True if all letters in word are in letter bank.
+# return False if word letter not in letter bank or word use letter more than it appears in letter bank.
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+    word_freq = {} # counting frequancy for each letter in word
+    for letter in word:
+        if letter in word_freq:
+            word_freq[letter] += 1
+        else:
+            word_freq[letter] = 1
+    
+    letter_bank_freq = {} # Count the frequency of each letter in the letter_bank
+    for letter in letter_bank:
+        letter = letter.upper()  
+        if letter in letter_bank_freq:
+            letter_bank_freq[letter] += 1
+        else:
+            letter_bank_freq[letter] = 1
+    
+    for letter in word_freq:
+        if letter not in letter_bank_freq or word_freq[letter] > letter_bank_freq[letter]:
+            return False
+    
+    return True
+
+
+score_chart =  {"A":1,
+                "B":3,
+                "C":3,
+                "D":2,
+                "E":1,
+                "F":4,
+                "G":2,
+                "H":4,
+                "I":1,
+                "J":8,
+                "K":5,
+                "L":1,
+                "M":3,
+                "N":1,
+                "O":1,
+                "P":3,
+                "Q":10,
+                "R":1,
+                "S":1,
+                "T":1,
+                "U":1,
+                "V":4,
+                "W":4,
+                "X":8,
+                "Y":4,
+                "Z":10
+                }
+# wave_03 :
+# we need to create dict that store letter scores.
+# then we calculate score that add corresponding freuancy to its letter.
+# return total score.
 
 def score_word(word):
-    pass
+    word = word.upper() # change to upper case to compare with score chart.
+    total_score = 0
+
+    word_dict = {} # count how many times letter appears.
+    for letter in word: #loop through each letter in word.
+        if letter in word_dict:
+            word_dict[letter] += 1
+        else:
+            word_dict[letter] = 1    
+
+    for letter, frequancy in word_dict.items(): # loop each letter to find its frequancy and score and sum scores up.
```

Comment:
If we go without the new dictionary you've created, we can absolutely accomplish this by simply looping through the word!